gateways/clients/wallet_service_db_client.py

```diff
@@ -105,10 +110,19 @@ def get_address_history(self, address: str, token: str, limit: int, offset: int)
 
         return result
 
-    def get_address_tokens(self, address: str, limit: int, offset: int) -> List[dict]:
+    def get_address_tokens(self, address: str, limit: int, offset: int) -> Tuple[int, List[dict]]:
         result: List[dict] = []
         found_htr: bool = False
+        total: int = 0
         with self.engine.connect() as connection:
+            cursor = connection.execute(text(address_tokens_count_query), address=address)
```

Comment:
Does SQL Alchemy protect us against SQL Injection when executing a raw query like this?

Reply:
SQLAlchemy quotes the parameters (escaping if needed).
The most recent version which is not fully protected against SQL Injection is 1.2.18 (more on this [here](https://security.snyk.io/vuln/SNYK-PYTHON-SQLALCHEMY-173678)) and we use 1.4.39.

Reply:
Great